File: src/app/Foundation.Template.Admin-UI/domain/models/permissionOrganisations/permissionOrganisationTypeInfos.ts
export class PermissionOrganisationTypeInfos {
    id: string;
    organisationTypeId: string;
    permissionLabel: string;
    permissionCode: string;
    permissionId: string;

    constructor(params: PermissionOrganisationTypeInfosDTO) {
        this.id = params.id;
        this.organisationTypeId = params.organisationTypeId;
        this.permissionLabel = params.permissionLabel;
        this.permissionCode = params.permissionCode;
        this.permissionId = params.permissionId;
    }
}

export interface PermissionOrganisationTypeInfosDTO {
    id: string;
    organisationTypeId: string;
    permissionLabel: string;
    permissionCode: string;
    permissionId: string;
}

export interface PermissionOrganisationTypeFilter {
    organisationTypeId?: string
    search?: string
}

export interface UpsertPermissionOrganisation {
    organisationTypeId: string;
    permissionIds: string[];
}

File: src/shared/Foundation.Extension.Domain/Models/PermissionOrganisations/PermissionOrganisationTypeInfos.cs
using System;
using System.Collections.Generic;

namespace Foundation.Extension.Domain.Models
{
    public class PermissionOrganisationTypeInfos
    {
        public Guid Id { get; set; }
        public Guid PermissionId { get; set; }
        public string PermissionCode { get; set; }
        public Guid OrganisationTypeId { get; set; }
        public string PermissionLabel { get; set; }
        public List<TranslationPermissionOrganisation> TranslationPermissions { get; set; }
    }
}